cmd/entrypoint/endpoint_routing.go

```diff
@@ -20,13 +20,21 @@ func makeEndpoints(ctx context.Context, ksnap *snapshot.KubernetesSnapshot, cons
 
 	result := map[string][]*ambex.Endpoint{}
 
-	for _, k8sEp := range ksnap.Endpoints {
-		svc, ok := k8sServices[key(k8sEp)]
-		if !ok {
-			continue
-		}
-		for _, ep := range k8sEndpointsToAmbex(k8sEp, svc) {
-			result[ep.ClusterName] = append(result[ep.ClusterName], ep)
+	svcEndpointSlices := map[string][]*kates.EndpointSlice{}
+
+	// Collect all the EndpointSlices for each service
+	for _, k8sEndpointSlice := range ksnap.EndpointSlices {
+		svcKey := fmt.Sprintf("%s:%s", k8sEndpointSlice.Namespace, k8sEndpointSlice.Labels["kubernetes.io/service-name"])
```

Comment:
Is it possible for the `kubernetes.io/service-name` label to be missing? (I don't think so, but if it somehow is, maybe we should just skip this object?)

Reply:
make sense, added check for label.